test/kubernetes_unittest.cc

```diff
@@ -68,4 +80,67 @@ TEST_F(KubernetesTest, GetNodeMetadata) {
   });
   EXPECT_EQ(big->ToString(), m.metadata.metadata->ToString());
 }
+
+TEST_F(KubernetesTest, ComputePodAssociations) {
+  const std::string api_version = "1.2.3";
+  const std::string kind = "TestKind";
+  const std::string uid1 = "TestUID1";
+  Configuration config(std::stringstream(
+    "KubernetesClusterName: TestClusterName\n"
+    "KubernetesClusterLocation: TestClusterLocation\n"
+    "MetadataIngestionRawContentVersion: TestVersion\n"
+    "InstanceZone: TestZone\n"
+    "InstanceId: TestID\n"
+  ));
+  Environment environment(config);
+  KubernetesReader reader(config, nullptr);  // Don't need HealthChecker.
+  const std::string encoded_ref = boost::algorithm::join(
+      std::vector<std::string>{api_version, kind, uid1}, "/");
+  json::value controller = json::object({
+          {"controller", json::boolean(true)},
+          {"apiVersion", json::string(api_version)},
+          {"kind", json::string(kind)},
+          {"name", json::string("TestName")},
+          {"uid", json::string(uid1)},
+	  {"metadata", json::object({
+	    {"name", json::string("InnerTestName")},
+	    {"kind", json::string("InnerTestKind")},
+	    {"uid", json::string("InnerTestUID1")},
+	  })},
+  });
+  InsertIntoOwners(encoded_ref, controller, &reader);
+  json::value pod = json::object({
+    {"metadata", json::object({
+      {"namespace", json::string("TestNamespace")},
+      {"uid", json::string("TestUID0")},
+      {"ownerReferences", json::array({
+        json::object({{"no_controller", json::boolean(true)}}),
+	controller->Clone(),
+      })},
+    })},
+    {"spec", json::object({
+      {"nodeName", json::string("TestSpecNodeName")},
+    })},
+  });
+
+  auto associations = ComputePodAssociations(pod->As<json::Object>(), reader);
+
+  EXPECT_EQ(associations->ToString(), json::object({
```

Comment:
I would suggest pulling this `json::object` out into an `expected_associations` local.